homeassistant/components/bayesian/binary_sensor.py

```diff
@@ -133,15 +133,15 @@ def __init__(self, name, prior, observations, probability_threshold, device_clas
                 to_observe.update(set([obs.get("entity_id")]))
             if "value_template" in obs:
                 to_observe.update(set(obs.get(CONF_VALUE_TEMPLATE).extract_entities()))
-        self.entity_obs = dict.fromkeys(to_observe, [])
+        self.entity_obs = {key: [] for key in to_observe}
 
         for ind, obs in enumerate(self._observations):
             obs["id"] = ind
             if "entity_id" in obs:
-                self.entity_obs[obs["entity_id"]].append(obs)
+                self.entity_obs.get(obs.get("entity_id")).append(obs)
```

Comment:
Let's keep this as `[]`. `get` has slightly different behavior in that it returns `None` instead of raising `KeyError`. Since we know it exists, we should use `[]`.

Reply:
Ah, I figured using get was better because of that, but I guess that'll fail silently and you want the exception? I'll put it back as it was originally.

Reply:
Yeah, failing silently is a pain, hard to find.

Reply:
Yeah, failing silently is a pain, hard to find.